### ISSUE TITLE
SA1000 does not work well with C# 9.0 target-typed new()

### ISSUE THREAD

**Issue by PaulBol** [DotNetAnalyzers/StyleCopAnalyzers#3561]
SA1000 demands putting a space after `new`. When using a target-typed `new` with C# 9.0 this leads to

```cs
List<int> xs = new ();
List<int> ys = new (capacity: 10_000);
List<int> zs = new () { Capacity = 20_000 };
```

This is an usual convention from my perspective and is not what https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/new-operator uses.

The SA1000 documentation states

> The `new` and `stackalloc` keywords should always be followed by a space, except in the following cases:
> 
> * The `new` or `stackalloc` keyword is used to create a new implicitly-typed array. In this case there should be no
>  space between the keyword and the opening array bracket.
> * The `new` keyword is part of a generic type constraint. In this case there should be no space between the `new`
>   keyword and the opening parenthesis.

I think `new()` should be a third case that does not require a space. The target-typed new seems very similar to an implicitly-typed array.

**Comment by jizc** [comment]
Are you using v1.1? This has been fixed in the v1.2.0 beta: [Update SA1000 for target-typed new expressions](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/commit/3874af7bcc5b3b79c628f824ea88c185bc5bc374)

**Comment by PaulBol** [comment]
Ok, I feel stupid now. I had specifically suggested to the coworker to use the latest beta. Seeing your feedback I checked to find that they had been using version 1.1.118. Sorry for the useless input.

Maybe https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md could be updated when 1.2 is released.
